query.go

```diff
@@ -72,7 +72,22 @@ func QueryOptionEDNS0(maxResponseSize uint16, flags int) QueryOption {
 	}
 }
 
-// NewQuery constructs a [*dns.Message] containing a query.
+// QueryOptionID allows setting an arbitrary query ID.
+//
+// Otherwise, the default is using [dns.Id] for all protocols
+// except DNS-over-HTTPS and DNS-over-QUIC, where we use
+// zero, thus following RFC 9250 Sect 4.2.1.
+func QueryOptionID(id uint16) QueryOption {
```

Comment:
Are we using this function anywhere used except in test cases?

Reply:
Ok, this is used to set an arbitrary value. We don't have that use case for now.

Reply:
Correct, and thank you for raising this point, which I think is very valid!

Under the YAGNI principle, we could not have implemented this. However, my thinking is that the functionality is small and self contained, and orthogonal with the rest of the codebase. 

Additionally, we have seen that upstream DoQ servers behave differently with respect to the query ID strictly being zero. Therefore, it seems potentially useful for fingerprinting. 

All in all, this seems to me to vouch in favour of adding it, since the cost is small and there is a potential benefit and use case (i.e., testing for compliance with the RFC).

Adding query option to `rbmk dig` to set this would be good.